src/pages/BurgerOrder/components/SelfMadeBurger/SelfMadeBurger.tsx

```diff
@@ -1,61 +1,100 @@
-import React from 'react';
+import React, { useState } from 'react';
 import * as customBurgerTypes from '../../../../common/models/customBurgerTypes';
-import { useSelector, useDispatch } from 'react-redux';
+import { useSelector } from 'react-redux';
 import { RootState } from '../../../../common/store/store';
-import { chooseBurgerBread } from '../../../../common/store/customBurgerStore/customBurgerActionCreators';
 import Ingredient from '../Ingredient/Ingredient';
+import BurgerBuilder from '../BurgerBuilder/BurgerBuilder';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { Button } from '@material-ui/core';
 import './selfMadeBurger.scss';
 
 const SelfMadeBurger = (): JSX.Element => {
-  const { bread, meat, vegetable, cheese, sauce } = useSelector((state: RootState) => ({
+  const { bread, meat, vegetable, cheese, sauce, price } = useSelector((state: RootState) => ({
     bread: state.customBurger.bread,
     meat: state.customBurger.meat,
     vegetable: state.customBurger.vegetable,
     cheese: state.customBurger.cheese,
     sauce: state.customBurger.sauce,
+    price: state.customBurger.totalPrice,
   }));
 
-  const dispatch = useDispatch();
+  const [buildState, setBuildState] = useState(1);
 
-  const handleBreadChange = (event: React.MouseEvent, breadName: customBurgerTypes.Bread): void => {
-    event.preventDefault;
+  const handleNextButton = (event: React.MouseEvent): void => {
+    event.preventDefault();
+    if (buildState < 5) {
+      setBuildState(buildState + 1);
+    }
+  };
 
-    if (bread !== breadName) {
-      dispatch(chooseBurgerBread(breadName));
+  const handlePrevButton = (event: React.MouseEvent): void => {
+    event.preventDefault();
+    if (buildState > 1) {
+      setBuildState(buildState - 1);
     }
   };
 
+  const burgerMeat = meat.map((ele: customBurgerTypes.Meat) => {
```

Comment:
if meat state is constructed with object, the logic should be something like
```suggestion
  const burgerMeat = Object.keys(state,meat).map(ele => {
      return new Array(ele.amount).map((, index) => {
          return <Ingredient ingType={ele} key={ele + index} />
       })
    })
```